### ISSUE TITLE
Make minimize on click work better when multiple windows of the same program are opened

### ISSUE THREAD

**Issue by andrei** [micheleg/dash-to-dock#728]
Minimize on click doesn't work well when there are multiple windows of the same program grouped together. I would want "showing previews" to be combined with minimize on click so that we don't lose this "showing previews" feature.

Ideally it should work like this:

* When clicking: if all windows are hidden, show all (or the last active one)
* When clicking: if windows are shown, show the previews
* When clicking: if the user clicks the icon again while the previews are shown, minimize all windows

This is how it used to work on Ubuntu Unity before 17.10 if minimize on click was enabled.

**Comment by andrei** [comment]
I noticed in the latest version there's the option "minimize or show previews". But this still doesn't cut it, and doesn't make it behave like Unity because for groups of windows the minimize action is effectively lost. See what I wrote above at this point:

* When clicking: if the user clicks the icon again while the previews are shown, minimize all windows

**Comment by micheleg** [comment]
Hi, 

we don't necessary need to replicate the behaviour of Unity, but what you describe seems a reasonable alternative behaviour. I haven't thought much about this option when I introduced it as it came as contribution (#711).

Unfortunately I don't think i have the time to dedicate to this now, but I would be happy to accept a concise patch improving the behaviour of this functionality. For people interested, if you look at the pull request (#711) you'll notice that the implementation is very self contained so it should not be too difficult to change it's logic according to the above design.

**Comment by wesaphzt** [comment]
Hi,

I've just switched to gnome and I was going to suggest similar to this.
It doesn't seem right to have to choose between 'minimize' and 'previews' setting.
My suggestion would have been to be able to combine having the preview of multiple windows along with minimizing single window apps, but the suggestions above are also good.

**Comment by dvorapa** [comment]
Yeah, the "minimize" part of this option does not work for me too
